### PR TITLE
[RW-5740][risk=no] Update e2e notebook tests

### PR DESCRIPTION
**Pull request by aweng98**
Changes in this PR includes:

- Break up `notebook-owner-actions.spec` test: I don't think it's needed because test requires a workspace and a Py kernel notebook. But test doesn't run any Py code. It normally takes 3 minutes to run. What this test covers can be covered "evenly" in other existing notebook tests. After removal of test, we need one less workspace and one less notebook.

- Add new `notebook-upload-file.spec` test: Watching Puppeteer type code in `nbstripoutput-filter.py` line by line taking a long time was painful. We didn't have test coverage for notebook file upload.

- Misc other changes that I hope will reduce flaky tests.

### CODE REVIEW

File: e2e/tests/notebook/notebook-create-python.spec.ts [hunk 1]
```diff
@@ -125,7 +127,38 @@ describe('Create python kernel notebook', () => {
 
     // Ideally we would validate the download URLs or download content here.
     // As of 9/25/20 I was unable to find a clear mechanism for accessing this.
+  });
+
+  test('Notebook name is unique', async () => {
```
Comment (aweng98) [inline e2e/tests/notebook/notebook-create-python.spec.ts:132]: This test is from `notebook-owner-actions.spec`.

File: e2e/tests/notebook/notebook-create-r.spec.ts [hunk 2]
```diff
@@ -49,7 +53,60 @@ describe('Create R kernel notebook', () => {
     });
     expect(cell3Output).toMatch(/success$/);
 
+    await notebook.save();
+  });
+
+  test('Duplicate rename delete notebook', async () => {
```
Comment (aweng98) [inline e2e/tests/notebook/notebook-create-r.spec.ts:59]: This test is from `notebook-owner-actions.spec`.

File: e2e/tests/ui/notebook-ui.spec.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+describe('Notebook and Runtime UI Test', () => {
+  beforeEach(async () => {
+    await signInWithAccessToken(page);
+  });
+
+  // This notebook is created by notebook-create-python.spec
+  const workspaceName = 'e2eCreatePythonKernelNotebookTest';
+  let pyNotebookName: string;
+
+  test('Notebook name is unique', async () => {
```
Comment (aweng98) [inline e2e/tests/ui/notebook-ui.spec.ts:19]: This test is from `notebook-owner-actions.spec`.

File: e2e/tests/notebook/notebook-create-python.spec.ts [hunk 4]
```diff
@@ -76,22 +76,15 @@ describe('Create python kernel notebook', () => {
 
     expect(
       await notebook.runCodeCell(4, {
-        codeFile: 'resources/python-code/nbstripoutput-filter.py',
-        markdownWorkaround: true
-      })
-    ).toMatch(/success$/);
```
Comment (aweng98) [inline e2e/tests/notebook/notebook-create-python.spec.ts:82]: Running `nbstripoutput-filter.py` is covered in `notebook-upload-file.spec`. This is going to make this test run much faster.

File: e2e/utils/test-utils.ts [hunk 5]
```diff
@@ -131,7 +137,6 @@ export async function createWorkspace(
   const workspacesPage = new WorkspacesPage(page);
   await workspacesPage.load();
   await workspacesPage.createWorkspace(workspaceName, cdrVersionName);
-  logger.info(`Created workspace "${workspaceName}" with CDR version "${cdrVersionName}"`);
```
Comment (aweng98) [inline e2e/utils/test-utils.ts:134]: redundant log.
Screen Shot 2021-09-10 at 12 43 28 PM

File: e2e/utils/test-utils.ts [hunk 6]
```diff
@@ -50,9 +50,15 @@ export async function signInWithAccessToken(page: Page, tokenFilename = config.U
   // logs; there is some delay between a console.log() execution and capture by
   // Puppeteer. Any console.log() within the above global function, for example,
   // is unlikely to be captured.
-  await homePage.reloadPage();
-  await homePage.gotoUrl(PageUrl.Home);
-  await homePage.waitForLoad();
+  try {
+    await homePage.reloadPage(2 * 60 * 1000);
+    await homePage.gotoUrl(PageUrl.Home);
```
Comment (aweng98) [inline e2e/utils/test-utils.ts:55]: `reloadPage` and `gotoUrl` functions could fail on rare occasions.

**Review by freemabd — APPROVED** [review]
Looks good.

**Pull request merged**